client/src/components/ToDoList/useTodosState.js

```diff
@@ -0,0 +1,88 @@
+import { useEffect, useState } from "react";
+import moment from "moment";
+
+const saveListToStorage = (list) => {
+    const listJson = JSON.stringify(list);
+    localStorage.setItem("todoList", listJson);
+};
+
+const sortOverallTodosByDate = (overallTodos) => {
+    let sorted = {};
+    Object.keys(overallTodos).sort().forEach(date => {
+        sorted[date] = overallTodos[date];
+    });
+    return sorted;
+};
+
+const sortSpecificTodoListByTime = (todos) => todos.sort((thisTodo, otherTodo) => thisTodo.time.localeCompare(otherTodo.time));
+
+function useTodosState(initialTodos, setIsAddingItem) {
+    const [todoList, setTodoList] = useState(initialTodos);
+
+    useEffect(() => {
+        const newTodaysDate = moment().format("YYYY-MM-D");
```

Comment:
this doesn't need to be in useEffect - it will rerendered every time the condition in useEffect is called. I would suggest to pull it out of the component